src/App.tsx

```diff
@@ -1,61 +1,144 @@
+import React, { useState } from 'react';
+import { TodoList } from './components/TodoList/TodoList';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+interface Todo {
+  id: number;
+  title: string;
+  completed: boolean;
+  userId: number;
+}
+
+const getMaxId = (todos: Todo[]): number => {
+  const todosOfId = todos.map(todo => todo.id);
+
+  return Math.max(...todosOfId);
+};
+
+const makeNormalizedTodos = (todoList: Todo[]) => {
+  return todoList.map(todo => {
+    const findUser = usersFromServer.find(user => user.id === todo.userId);
+
+    return { ...todo, user: findUser };
+  });
+};
 
 export const App = () => {
+  const [todos, setTodos] = useState([...todosFromServer]);
+  const [inputValue, setInputValue] = useState('');
+  const [hasTitle, setHasTitle] = useState(true);
+  const [userId, setUserId] = useState(0);
+  const [isUserOption, setUserOption] = useState(true);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(event.target.value);
+
+    if (event.target.value) {
+      setHasTitle(true);
+    }
+  };
+
+  const handleSelectionChange = (
+    event: React.ChangeEvent<HTMLSelectElement>,
+  ) => {
+    setUserId(+event.target.value);
+
+    if (event.target.value) {
+      setUserOption(true);
+    }
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!inputValue.length || !userId) {
+      setHasTitle(Boolean(inputValue.length));
+      setUserOption(Boolean(userId));
+
+      return;
+    }
+
+    if (inputValue.length || userId) {
```

Comment:
you already check it
```suggestion
```

Reply:
you already check this here
![image](https://user-images.githubusercontent.com/91411692/235476128-d6b1bdb0-f1cf-4437-a51a-2dcc9dcc0b7e.png)